### ISSUE TITLE
metadata: regression in 'disable-push' handling

### ISSUE THREAD

**Issue by lucab** [sunng87/cargo-release#280]
I've just upgraded to latest `cargo-release` v0.16.1 and unfortunately I'm hitting a regression somewhere in the logic handling the `disable-push` option.

We do have a very simple project (one binary crate, no libs, no workspaces) available in [this public repository](https://github.com/coreos/zincati/tree/main).
Its `Cargo.toml` has the following metadata for doing release (in two steps via manual PRs and reviews):
```
$ tail -8 Cargo.toml 
[package.metadata.release]
disable-publish = true
disable-push = true
post-release-commit-message = "cargo: development version bump"
pre-release-commit-message = "cargo: zincati release {{version}}"
sign-commit = true
sign-tag = true
tag-message = "zincati {{version}}"
```

This config has always worked fine so far. However, using the latest `cargo-release` release it looks like `disable-push = true` is now suddenly ignored, which resulted in both branch and tag being pushed to the git remote:
```
$ cargo release
[2021-07-07T09:01:24Z WARN ] release-0.0.22 is behind origin/release-0.0.22
Release zincati 0.0.22? [y/N] 
y
[2021-07-07T09:02:15Z INFO ] Update zincati to version 0.0.22
[release-0.0.22 7e614af] cargo: zincati release 0.0.22
 2 files changed, 2 insertions(+), 2 deletions(-)
[2021-07-07T09:02:17Z INFO ] Starting zincati's next development iteration 0.0.23-alpha.0
[release-0.0.22 13929c6] cargo: development version bump
 2 files changed, 2 insertions(+), 2 deletions(-)
[2021-07-07T09:02:18Z INFO ] Pushing v0.0.22 to origin
Enumerating objects: 8, done.
Counting objects: 100% (8/8), done.
Delta compression using up to 8 threads
Compressing objects: 100% (5/5), done.
Writing objects: 100% (5/5), 1.78 KiB | 1.78 MiB/s, done.
Total 5 (delta 3), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
To github.com:coreos/zincati.git
 * [new tag]         v0.0.22 -> v0.0.22
[2021-07-07T09:02:22Z INFO ] Pushing HEAD to origin
fatal: The current branch release-0.0.22 has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin release-0.0.22

```

This is problematic as it breaks/regresses existing configurations, and results in a tag/branch pushing which is not meant to happen at that point.

**Comment by epage** [comment]
Yes, this was broken as part of some workspace work.  For a while, push was crate-specific.  We then moved it to being exclusively from workspace config, which broke things for single-crate users.  We then changed things up to support both single crate and workspace but missed `disable-push`.

The line in question is https://github.com/sunng87/cargo-release/blob/master/src/main.rs#L844

A workaround, until this is fixed, is to put `disable-push = true` in `release.toml` instead of `Cargo.toml`.
